traffic_ops/traffic_ops_golang/routing/routes.go

```diff
@@ -133,6 +134,12 @@ func Routes(d ServerData) ([]Route, []RawRoute, http.Handler, error) {
 		 * 4.x API
 		 */
 
+		// CDN lock
+		{api.Version{Major: 4, Minor: 0}, http.MethodGet, `cdn_locks/?$`, cdn_lock.Read, auth.PrivLevelReadOnly, Authenticated, nil, 4134390561},
+		{api.Version{Major: 4, Minor: 0}, http.MethodPost, `cdn_locks/?$`, cdn_lock.Create, auth.PrivLevelOperations, Authenticated, nil, 4134390562},
+		{api.Version{Major: 4, Minor: 0}, http.MethodDelete, `cdn_locks/?$`, cdn_lock.Delete, auth.PrivLevelOperations, Authenticated, nil, 4134390564},
+		{api.Version{Major: 4, Minor: 0}, http.MethodDelete, `cdn_locks/admin/?$`, cdn_lock.AdminDelete, auth.PrivLevelAdmin, Authenticated, nil, 4134390565},
```

Comment:
So here's what I was thinking: If an `admin` level user hits the `DELETE` endpoint, it might be because the user wants to delete their own lock, or some other user's lock. That why I wanted to split them out into two different endpoints, one for deleting the user's own locks, and one for other deleting other people's locks (for an admin). However, I could add another query parameter to the DELETE endpoint, saying something like `self=true` to imply that the user wants to delete their own locks. Thoughts?

Reply:
The primary key for a CDN lock is the CDN which it locks, so it's impossible to for more than one user to have a lock on a CDN at a time. the `DELETE` method requires the name of the CDN to be passed in the `cdn` query string parameter according to the documentation. So the request already fully and uniquely defines the lock to be deleted, so passing `self=true` or `self=false` doesn't make much sense to me. The lock either belongs to you or it doesn't, you can't delete only "your" lock on a CDN because a CDN only has one lock. The only thing that needs to be decided is whether or not the requesting user has permission to perform the operation.

Reply:
Agreed, but that does leave us to the chance of an admin accidentally or unintentionally deleting someone else's lock. I know we could pass it off as user error, but I was just thinking of adding some more safety into it. What do you think?

Reply:
Well that's what I mean, the admin user will say "I'm going to unlock this CDN" and then... that's exactly what happens. I don't think it's super likely to accidentally delete someone else's lock, because if someone else has it locked then you couldn't have locked it in the first place. So if you have no memory of locking the CDN, you should be able to immediately suppose that the lock was placed by someone else.

If we're worried about users doing legal things accidentally, then I'd say the proper way to handle that is to place a warning and/or confirmation dialog in the UI. But IMO we don't need a whole separate endpoint to describe a second way of checking for permissions to do a single operation. 1 operation <-> 1 method of one endpoint makes the most sense to me.

Reply:
Sounds good, just wanted to put it out there. 👍